### ISSUE TITLE
Add an ecosystem/generators/plugins page

### ISSUE THREAD

**Issue by chasenlehara** [donejs/donejs#682]
__tldr;__ Create a page on donejs.com that displays __plugins__ and __generators__ created as part of the DoneJS ecosystem.

> This was discussed on a [recent live stream (53:31)](https://youtu.be/ieZJrPwSZYQ?t=53m31s) and at a [contributors meeting (43:35)](https://youtu.be/95OX6-qii24?t=43m35s)

# The Problem

DoneJS has an enthusiastic ecosystem of developers! Well, this is not the problem, the problem is we have done a poor job of sharing their work. Way back when we had a portal where users could share their projects and vote on others; a Reddit for DoneJS (then called JavaScriptMVC) stuff, if you will. Alas, that fell by the wayside and today we don't have a good way of sharing people's awesome projects.

This means:

* If someone builds a cool can-component they have to rely on word-of-mouth to get others to use it.
* If you are searching for a component for your needs, you might not find it and rebuild it yourself.

There are entire GitHub organizations dedicated to DoneJS plugins that most people don't even know about, like this one:

![screen shot 2017-09-22 at 8 03 05 am](https://user-images.githubusercontent.com/361671/30743631-6d00039a-9f6c-11e7-9861-002c51cb9433.png)

# The Solution

We can do better! Let's add a page to donejs.com, under the __Community__ heading, that provides a list of community generators and plugins.

## Design

The design should be simple, but can consist of:

* The name of the plugin/generator
* The author
* A version number

I would mirror the design from similar sites like github.com and npmjs.org. Here's what npm's listing looks like:

<img width="718" alt="screen shot 2017-09-22 at 8 13 05 am" src="https://user-images.githubusercontent.com/361671/30743949-d1348df8-9f6d-11e7-8f9e-79dd5bb50064.png">

## Contributing

Learning from the past attempts, let's make it easy for users to contribute new generators and plugins for the listing. I propose having a __donejs-ecosystem__ repository that houses these lists.

Using a markdown format list is the easiest way for people to contribute. It should be as easy as editing a markdown file like so:

```markdown
# Plugins
...
* [my-awesome-component](https://github.com/my-user/my-awesome-component)
```

# Implementation

Now that we have bit-docs, it should be fairly simple to create a plugin that:

* Given the list of generators/plugins, queries npm and/or GitHub for metadata about the project
* Generates the HTML listing.

This should probably be implemented as a build-time plugin since that is your best change to query the server. Some parts, like the version number, might be better suited to be dynamically rendered in the client.

The goal of the implementation should be simplicity, however, so if using a bit-docs plugin proves to be too complex, then we can require more of this information from the user as part of the submission process.

# Tasks

* [ ] - Get designs for what this page will look like, and what each listing should look like.
* [ ] - Decide on where the PRs should be submitted to. A new repo would be easier for users, but might sacrifice the continuous deployment we have with the donejs/donejs repo.
* [ ] - Implement any bit-docs plugins, if we decide to go that way.
* [ ] - Backfill with known community generators/plugins (with the author's permission).
* [ ] - Write up a blog article announcing the new ecosystem portal.

**Comment by matthewp** [comment]
👍 I think the simplest way is to just have a list in markdown and allow people to submit PRs to add theirs to the list.

**Comment by nriesco** [comment]
@matthewp meanwhile here's my contribution: configurable typeahead (based on twitter's typeahead) [https://github.com/riescorp/donejs-typeahead](https://github.com/riescorp/donejs-typeahead)

**Comment by matthewp** [comment]
Awesome @nriesco !  That looks really nice.

**Comment by frank-dspeed** [comment]
We Need a Global Place like npm and atom for all donejs plugins
and we need the donejs package manager

- META REPOSITORY we can use github as database via hosting json files as maps to the files.
- We Should split the Meta Repository Like Docker does it its a good example for some 100k of images
its always a composition out of 2 repos the src repo and the MANIFEST Repo that holds the references to all 
- https://github.com/docker-library/mariadb
- https://github.com/docker-library/official-images/blob/master/library/mariadb
so https://github.com/docker-library/official-images is the meta repo it 

so i created organization donejs-library and simply will add all existing plugins that i find and build some instructions how to add plugins then we have the ability to create a ghpage or else with all plugins and generators

i created offical-plugins and offical-generators repo so that we can control what generatoras are considered offical i think there are some rules that need to get enforces so we make sure no generator from a 3rd party brakes existing functionality as also the same for plugins like a clear donejs version they are made for.

**Comment by matthewp** [comment]
Going to remove this from the 1.0 epic as I don't think it will be part of that.

**Comment by andrejewski** [comment]
New projects created with `generator-donejs` will contain a `donejs-{app|plugin|generator}` keyword via https://github.com/donejs/generator-donejs/pull/233. So now we can start curating DoneJS resources via the following tags: `donejs-app`, `donejs-plugin`, and `donejs-generator`.
